### PR TITLE
Using GetUserByEmailContext to query user by email

### PR DESCRIPTION
**Pull request by billcchung**
# Summary
Update the user query to use [slack client](https://github.com/slack-go/slack) `GetUserByEmailContext` https://github.com/slack-go/slack/blob/master/users.go#L396 when query type is `email`

# Background

We're currently implementing adding slack usergroups for services, and found the way this provider gets the user by emails is by retrieving the whole list of users, we understand it caches the list as file, but we're running terraform inside CI, so the cache file isnt ideal for us.

Fortunately slack already provided an API to retrieve specific user with email `GetUserByEmailContext` (https://github.com/slack-go/slack/blob/master/users.go#L396), and underlaying it's using [users.lookupByEmail](https://api.slack.com/methods/users.lookupByEmail) api call, which is [tier3](https://api.slack.com/docs/rate-limits#tier_t3) rate limiting, it's better than [users.list](https://api.slack.com/methods/users.list) that has tier2 rate limiting.

I think it's better to retrieve specific user rather than the whole list (when possible), because it reduces the number of necessary calls as well as the API has higher rate limiting, which end up with better scalability. (we have quite some services which will run CI individually.)

# Verification

I've built one binary and verified it works:
```
  # module.<service>.slack_usergroup.slack_usergroup[0] will be created
  + resource "slack_usergroup" "slack_usergroup" {
      + handle  = "<handle>"
      + id      = (known after apply)
      + name    = "<name>"
      + team_id = (known after apply)
    }

  # module.<service>.slack_usergroup_members.slack_usergroup_members[0] will be created
  + resource "slack_usergroup_members" "slack_usergroup_members" {
      + id           = (known after apply)
      + members      = [
          + "<memberID>",
        ]
      + usergroup_id = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.
```

Note: I also put a line to log while testing, to really make sure it's using the new call `GetUserByEmailContext`, but didnt put it in the PR.

### CODE REVIEW

**Comment by billcchung** [comment]
@jmatsu would you mind taking a look of this and see if it's appropriate? I've added some background as well as verifications

**Comment by billcchung** [comment]
hi @jmatsu , sorry to ping you again on this? it's currently blocking me, not sure if you have time to take a look on this, please also feel free to let me know if I need to change anything, thanks!

**Comment by jmatsu** [comment]
Sorry, i haven't noticed your mentions. I'll check this within several hours.

File: slack/data_user.go [hunk 1]
```diff
@@ -19,7 +24,7 @@ func dataSourceSlackUser() *schema.Resource {
 			"query_type": {
 				Type:         schema.TypeString,
 				Required:     true,
-				ValidateFunc: validateEnums([]string{"id", "name", "email"}),
+				ValidateFunc: validateEnums([]string{userQueryTypeID, userQueryTypeName, userQueryTypeEmail}),
```
Comment (jmatsu) [inline slack/data_user.go:27]: 👍

**Review by jmatsu — APPROVED** [review]
LGTM. Thank you for your contribution!

**Comment by jmatsu** [comment]
please note that v0.6.2 includes this change.

**Pull request merged**